src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -310,10 +311,71 @@ def _process_resource_change(
                 data[name] = serialize(instance)
 
             for installation in installations:
-                # Trigger a new task for each webhook
-                send_resource_change_webhook.delay(
-                    installation_id=installation.id, event=str(event), data=data
-                )
+
+                if _is_project_allowed(installation, instance.project_id):
+                    # Trigger a new task for each webhook
+                    send_resource_change_webhook.delay(
+                        installation_id=installation.id, event=str(event), data=data
+                    )
+
+
+def _is_project_allowed(installation: RpcSentryAppInstallation, project_id: int) -> bool:
+    service_hook = _load_service_hook(installation.organization_id, installation.id)
+    if not service_hook:
+        logger.info("send_webhooks.missing_servicehook", extra={"installation_id": installation.id})
+        return False
+    # must use 2 separate helpers for cache invalidation to work correctly in [] <-> [project1, ...] scenarios
+    return not _is_project_filtering_enabled(service_hook.id) or _does_project_filter_allow_project(
+        service_hook.id, project_id
+    )
+
+
+@cache_func_for_models(
+    [
+        (
+            ServiceHook,
+            lambda service_hook: (service_hook.organization_id or -1, service_hook.actor_id or -1),
+        )
+    ],
+    recalculate=False,
+)
+def _load_service_hook(organization_id: int, installation_id: int) -> ServiceHook | None:
+    try:
+        service_hook = ServiceHook.objects.get(
+            organization_id=organization_id,
+            actor_id=installation_id,
+        )
+        if service_hook.installation_id != service_hook.actor_id:
+            logger.error(
+                "service_hook.installation_id != service_hook.actor_id",
+                extra={"service_hook": service_hook.id},
+            )
+        return service_hook
+    except ServiceHook.DoesNotExist:
+        return None
+
+
+@cache_func_for_models(
+    [(ServiceHookProject, lambda hook_project: (hook_project.service_hook,))],
```

Comment:
This should be `service_hook_id`, since that's what we use to call this function. Maybe doesn't matter too much, since we've set `recalculate=False`.

Reply:
oh thanks, fixed